unittests/test_loader.py

```diff
@@ -58,3 +59,8 @@ def test_conflicted_checks(self):
     def test_load_error(self):
         self.assertRaises(OSError, self.loader.load_from_file,
                           'unittests/resources/checks/foo.py')
+
+    def test_load_invalid_sytax(self):
+        self.assertRaises(ReframeSyntaxError, self.loader.load_from_file,
+                          'unittests/resources/checks_unlisted/'
+                          'invalid_syntax_check.py')
```

Comment:
Here you should check that the `filename` attribute of the raised `SyntaxError` is set properly.